src/layouts/Layout.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Outlet } from "react-router-dom";
+import SearchAppBar from "../components/Appbar";
+import Stack from "@mui/material/Stack";
+import Box from "@mui/material/Box";
+
+export const Layout = () => {
+  return (
```

Comment:
Should useEffect to check the condition to check authentication permission. If not, redirect to homepage.